consensus/hotstuff/committees/consensus_committee_test.go

```diff
@@ -16,6 +17,57 @@ import (
 	"github.com/onflow/flow-go/utils/unittest/mocks"
 )
 
+// TestConsensus_InvalidSigner tests that the appropriate sentinel error is
+// returned by the hotstuff.Committee implementation for non-existent or
+// non-committee identities.
+func TestConsensus_InvalidSigner(t *testing.T) {
+
+	realIdentity := unittest.IdentityFixture(unittest.WithRole(flow.RoleConsensus))
+	realNonCommitteeIdentity := unittest.IdentityFixture(unittest.WithRole(flow.RoleConsensus), unittest.WithStake(0))
+	fakeID := unittest.IdentifierFixture()
+	blockID := unittest.IdentifierFixture()
+
+	state := new(protocolmock.State)
+	snapshot := new(protocolmock.Snapshot)
+
+	// create a mock epoch for leader selection setup in constructor
+	currEpoch := newMockEpoch(
+		1,
+		unittest.IdentityListFixture(10),
+		1,
+		100,
+		unittest.SeedFixture(32),
+	)
+	epochs := mocks.NewEpochQuery(t, 1, currEpoch)
+	snapshot.On("Epochs").Return(epochs)
+
+	state.On("Final").Return(snapshot)
+	state.On("AtBlockID", blockID).Return(snapshot)
+
+	snapshot.On("Identity", realIdentity.NodeID).Return(realIdentity, nil)
+	snapshot.On("Identity", realNonCommitteeIdentity.NodeID).Return(realNonCommitteeIdentity, nil)
+	snapshot.On("Identity", fakeID).Return(nil, protocol.IdentityNotFoundError{})
+
+	com, err := NewConsensusCommittee(state, unittest.IdentifierFixture())
+	require.NoError(t, err)
+
+	t.Run("non-existent identity should return ErrInvalidSigner", func(t *testing.T) {
+		_, err := com.Identity(blockID, fakeID)
+		require.True(t, errors.Is(model.ErrInvalidSigner, err))
+	})
+
+	t.Run("existent but non-committee-member identity should return ErrInvalidSigner", func(t *testing.T) {
+		_, err := com.Identity(blockID, realNonCommitteeIdentity.NodeID)
+		require.True(t, errors.Is(model.ErrInvalidSigner, err))
+	})
+
```

Comment:
Thank you for extending the tests. While we check that:
* nodes with zero staked are filtered out

I think there are more edge-cases, which we could check:
* nodes with role _other_ than `RoleConsensus` are filtered out 
* ejected consensus nodes are filtered out

Reply:
Added cases in https://github.com/onflow/flow-go/pull/565/commits/dabbbfeda1d65102dcb288af356c1933168b2e99